### PR TITLE
Add rbac tests for service discovery.

### PR DESCRIPTION
**Pull request by khushboo-rancher**
Added create/edit/delete tests for service discovery.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_service_discovery.py [hunk 1]
```diff
@@ -348,6 +408,106 @@ def test_dns_record_type_selector():
...
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    workload = rbac_get_workload()
+    p_client = get_project_client_for_token(project, token)
+
+    record = {"type": "dnsRecord", "targetWorkloadIds": [workload["id"]],
+              "name": random_test_name("record"), "namespaceId": ns.id}
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+        with pytest.raises(ApiError) as e:
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_service_discovery.py]: We can simplify this:

```
if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
        with pytest.raises(ApiError) as e:
            dns_record = create_dns_record(record, p_client)
        assert e.value.error.status == 403
        assert e.value.error.code == 'Forbidden'
    else:
        dns_record = create_dns_record(record, p_client)
        # test dns record for the workload
        validate_dns_record_for_workload(workload, 1,
                                         record, p_client=p_client)
        p_client.delete(dns_record)

File: tests/validation/tests/v3_api/test_service_discovery.py [hunk 2]
```diff
@@ -348,6 +408,106 @@ def test_dns_record_type_selector():
...
+    """
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    workload = rbac_get_workload()
+    p_client = get_project_client_for_token(project, token)
+
+    record = {"type": "dnsRecord", "targetWorkloadIds": [workload["id"]],
+              "name": random_test_name("record"), "namespaceId": ns.id}
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_service_discovery.py]: We can simplify this here and in the other cases:

```
with pytest.raises(ApiError) as e:
    dns_record = create_dns_record(record, p_client)

if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
    assert e.value.error.status == 403
    assert e.value.error.code == 'Forbidden'
    return


validate_dns_record_for_workload(workload, 1,
                                 record, p_client=p_client)
p_client.delete(dns_record)
```

File: tests/validation/tests/v3_api/test_service_discovery.py [hunk 3]
```diff
@@ -1,12 +1,64 @@
...
+Test requirement:
+Below Env variables need to set
+CATTLE_TEST_URL - url to rancher server
+ADMIN_TOKEN - Admin token from rancher
+USER_TOKEN - User token from rancher
+RANCHER_CLUSTER_NAME - Cluster name to run test on
+RANCHER_TEST_RBAC - Boolean (Optional), To run role based tests.
+"""
+
+from .common import CLUSTER_OWNER
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_service_discovery.py]: Alphabetize imports

File: tests/validation/tests/v3_api/test_service_discovery.py [hunk 4]
```diff
@@ -371,6 +531,26 @@ def create_dns_record(record, p_client=None):
     return created_record
 
 
+def validate_dns_record_deleted(client, dns_record, timeout=DEFAULT_TIMEOUT):
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_service_discovery.py]: Move to `common.py`

File: tests/validation/tests/v3_api/test_service_discovery.py (hunk 2 above)
Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/test_service_discovery.py]: Raising error without condition can fail for cases where error is not expected.

File: tests/validation/tests/v3_api/test_service_discovery.py (hunk 3 above)
Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/test_service_discovery.py]: Done

File: tests/validation/tests/v3_api/common.py [hunk 5]
```diff
@@ -968,6 +968,26 @@ def dig_check():
...
 
 
+def validate_dns_record_deleted(client, dns_record, timeout=DEFAULT_TIMEOUT):
+    """
+    Checks whether ingress got deleted successfully.
+    Validates if ingress is null in for current object client.
+    @param client: Object client use to create ingress
+    @param dns_record: record object subjected to be deleted
+    @param timeout: Max time to keep checking whether ingress is deleted or not
+    """
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/common.py]: FIx the comments to indicate that this method works on dns_records

Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/common.py]: Done

**Pull request merged**